### ISSUE TITLE
Retain previously uploaded records/data versions

### ISSUE THREAD

**Issue by saragon02** [galterlibrary/InvenioRDM-at-NU#61]
As part of the version history, are previous record/data upload versions retained for access by the contributor?

**Comment by saragon02** [comment]
> Any of these versions of the record will be referenceable and kept.

Are they kept and available to the record creator, but **not** available to/downloadable by anyone else (except creator's proxy)?

**Comment by fenekku** [comment]
Apriori, everything is still accessible by everyone who had access to it before.

When changes (metadata changed and new version manually made or files replaced/deleted -- only possible with a new version) are made the previous version is kept with all the metadata and files it had. The concept DOI will point to the newest version, so the previous versions are a little bit more hidden. All these versions are accessible and files are downloadable to anyone who has the permission.

Deletion is a tricky issue because the goal of the repository is long term preservation. We *want* to keep previous versions and have those previous versions always consultable because permanent urls are a guarantee. There are times when a file does need to be expunged (e.g., copyright or privacy or objectionable content) completely. This is only possible for admins and careful thought should go into that decision. This is possible now, but there is no going back, so one has to be careful.

**Comment by saragon02** [comment]
> All these versions are accessible and files are downloadable to anyone who has the permission.

Thanks for the confirmation. And "anyone who has the permission" means any user based on the accessibility levels that were set (e.g., anyone if it was open access, or any NU user if accessibility was to the institution only, etc.), correct?

**Comment by fenekku** [comment]
That's correct. And I believe the permissions for each version is respected (since permissions are metadata too)
